engine/src/game_error.rs

```diff
@@ -0,0 +1,35 @@
+use crate::game_result::GameResult;
+
+#[derive(thiserror::Error, Debug)]
+pub enum GameError {
+    #[error("Invalid or illegal move on turn {turn}, moving piece {piece} from {from} to {to}")]
+    InvalidMove {
+        piece: String,
+        from: String,
+        to: String,
+        turn: usize,
+        reason: String,
+    },
+    #[error("Invalid spawn of piece {piece} at position {position} on turn {turn}")]
+    InvalidSpawn {
+        piece: String,
+        position: String,
+        turn: usize,
+        reason: String,
+    },
+    #[error("Invalid bug {bug:?}")]
+    InvalidBug { bug: String },
+    #[error("Invalid color {color:?}")]
+    InvalidColor { color: String },
+    #[error("Invalid game {gametype:?} type")]
+    InvalidGameType { gametype: String },
```

Comment:
nitpick: i think these could all be considered the same category of error, i.e. `ParsingError`, with specific messages for what the parsing issue was ("invalid bug", etc)

Reply:
done